### PR TITLE
parser for monitoring zmq errors on aci leaf switches

### PR DESCRIPTION
**Pull request by saptbell**
adding parser for ssh exporter to monitor tx/rx errors captured via command 'show system internal epm counters zmq' on aci leaf switches

### CODE REVIEW

File: prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl [hunk 1]
```diff
@@ -17,6 +17,29 @@ lookup_sources:
       router_project: SELECT DISTINCT id, project_id FROM neutron.routers
 
 metrics:
+  zmq_rx_error:
+    regex: >-
+      ^ZmQ\:\s*RX\sCNT\:\s\d+\,\sBYTES\:\s\d+\,\sERRORS\:\s(\d*)
+    multi_value: true
+    value: $1
+    labels:
+      errors: $1
```
Comment (swagner-de) [inline prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl:26]: Vector label and value are assigned to the same match group, whenever this counter increases a new vector with a new label is created. That would just blow up the count of vectors.

File: prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl [hunk 2]
```diff
@@ -17,6 +17,29 @@ lookup_sources:
...
 metrics:
+  zmq_rx_error:
+    regex: >-
+      ^ZmQ\:\s*RX\sCNT\:\s\d+\,\sBYTES\:\s\d+\,\sERRORS\:\s(\d*)
+    multi_value: true
+    value: $1
+    labels:
+      errors: $1  
+    description: check for zmq rx errors
+    metric_type_name: gauge
```
Comment (swagner-de) [inline prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl:28]: It looks like a counter?

**Review by swagner-de — CHANGES_REQUESTED** [review]
What do you think about this metric:
```yaml
  zmq_errors:
    regex: >-
      ^ZmQ\:\s*(RX|TX)\sCNT\:\s\d+\,\sBYTES\:\s\d+\,\sERRORS\:\s(\d*)
    multi_value: true
    value: $2
    labels:
      direction: $1
    description: check for zmq  errors
    metric_type_name: counter
    command: show system internal epm counters zmq
    timeout_secs: 5
```

It's a single metric that will create 2 vectors with the label `direction` representing `rx` and `tx`

```
# HELP ssh_zmq_errors check for zmq  errors
# TYPE ssh_zmq_errors counter
ssh_zmq_errors{direction="RX",host="10.17.125.118:22"} 0
ssh_zmq_errors{direction="TX",host="10.17.125.118:22"} 0
```

File: prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl (hunk 1 above)
Comment (tomas-ronec) [reply prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl:26]: you are right

File: prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl (hunk 2 above)
Comment (tomas-ronec) [reply prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl:28]: yes, counter would be better

**Comment by tomas-ronec** [comment]
the proposed metric from you @swagner-de  looks good to me as well. Can you make the change directly on this file in this PR? Thanks

File: prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl (hunk 2 above)
Comment (saptbell) [reply prometheus-exporters/network-generic-ssh-exporter/templates/_config.yaml.tpl]: Suggested changes have been made to _config.yaml.tpl

**Pull request merged**